src/main/java/org/opensearch/sdk/handlers/ExtensionsInitRequestHandler.java

```diff
@@ -60,6 +63,15 @@ public InitializeExtensionsResponse handleExtensionInitRequest(InitializeExtensi
                 extensionsRunner.opensearchNode,
                 extensionsRunner.getUniqueId()
             );
+            // Get OpenSearch Settings and set values on ExtensionsRunner
+            Settings settings = extensionsRunner.sendEnvironmentSettingsRequest(extensionTransportService);
+            extensionsRunner.setEnvironmentSettings(settings);
+            extensionsRunner.setNamedXContentRegistry(
+                new ExtensionNamedXContentRegistry(settings, extensionsRunner.getCustomNamedXContent())
+            );
+
```

Comment:
Can we add a TODO here for other extension points like createComponent? Do we have to set getSettings, getRestHandlers too here? Confused a little

Reply:
Don't think we need a TODO but I can definitely add more comments.  There's two categories here:
 - things the ExtensionRunner already knows about because they were on the Extension interface (getSettings, getRestHandlers) and just needs to send to OpenSearch
 - things we don't have enough information about yet (we need environment settings in order to create the NamedXContent registry)

Reply:
That helps